var/spack/repos/builtin/packages/p4est/package.py

```diff
@@ -48,8 +48,9 @@ class P4est(AutotoolsPackage):
           when='@2.0')
 
     def autoreconf(self, spec, prefix):
-        bootstrap = Executable('./bootstrap')
-        bootstrap()
+        if self.spec.satisfies('@:2.2'):
+            bootstrap = Executable('./bootstrap')
+            bootstrap()
```

Comment:
The error message you mentioned in https://github.com/spack/spack/pull/23824#issuecomment-868675657 is interesting:

> It is NOT required to run bootstrap to build from a tar.gz archive

So this seems like the best approach for the latest version of `p4est` (I would guess both 2.3.x).